Headers/DebugServer2/Target/ThreadBase.h

```diff
@@ -57,6 +57,9 @@ class ThreadBase {
                            Address const &address = Address()) = 0;
 
 public:
+  virtual ErrorCode beforeResume();
```

Comment:
The reason why it isn't in `resume`, though, is that the `resume` function is implemented in the platform-specific classes, not in the base class. And before-resume should always happen, regardless of platform. I suppose we could add a `resume` in the base class, which calls `beforeResume` and then the `resume` in the platform specific class, but I don't know that this is necessary/better